### PR TITLE
NO-JIRA: adm release: cleanups

### PR DESCRIPTION
**Pull request by petr-muller**
While working towards fixing [OCPBUGS-19824](https://issues.redhat.com/browse/OCPBUGS-19824) and [OCPBUGS-30080](https://issues.redhat.com/browse/OCPBUGS-30080) I needed to familiarize myself with this code. Taking opportunities for small improvements as I go.

Separating commits for easy review (and to allow easy removing some changes if not desired for any reason), but I think we can squash-merge them:

- **adm release: remove unused code**
- **adm release: port deprecated functions**
- **adm release: minor cleanups**
- **adm release: compile regex just once**
- **adm release: simplify regexes**

### CODE REVIEW

File: pkg/cli/admin/release/new.go
Comment (ardaguclu) [inline pkg/cli/admin/release/new.go:1500]: Isn't this function used by anywhere anymore?. Do you happen to know why?.

**Review by ardaguclu — COMMENTED** [review]
Code is much nicer than before. Dropped a question.

File: pkg/cli/admin/release/new.go
Comment (petr-muller) [reply pkg/cli/admin/release/new.go:1500]: I just grepped `git grep pruneEmptyDirectories` but yeah, let me grep the git history to get more context.

/hold

For context why I'm digging into this code, I'm looking into [OCPBUGS-19824](https://issues.redhat.com/browse/OCPBUGS-19824) and [OCPBUGS-30080](https://issues.redhat.com/browse/OCPBUGS-30080). I'll have more PRs like this.

Comment (petr-muller) [reply pkg/cli/admin/release/new.go:1500]: Haha, brought in by Clayton in 2018 (62d9e2c6358291547b9282a9e8f37f8f6681e4d4)... unused :rofl: 

```console
$ git log -S pruneEmptyDirectories
commit 773d900edd34c706efee594eda94d45bebab6cb1
Author: Petr Muller <muller@redhat.com>
Date:   Thu Nov 14 13:05:36 2024 +0100

    adm release: remove unused code

commit 62d9e2c6358291547b9282a9e8f37f8f6681e4d4
Author: Clayton Coleman <ccoleman@redhat.com>
Date:   Tue Jul 31 13:46:50 2018 -0400

    Add a prototype release command
    ...

$ git checkout 62d9e2c6358291547b9282a9e8f37f8f6681e4d4 --quiet
$ git grep pruneEmptyDirectories
pkg/oc/cli/admin/release/new.go:func pruneEmptyDirectories(dir string) error {
```

/hold cancel

**Comment by petr-muller** [comment]
https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oc/1914/pull-ci-openshift-oc-master-e2e-aws-ovn-serial/1857056629850116096
```
INFO[2024-11-14T17:38:24Z] Running step e2e-aws-ovn-serial-ipi-deprovision-deprovision. 
{"component":"entrypoint","file":"sigs.k8s.io/prow/pkg/entrypoint/run.go:169","func":"sigs.k8s.io/prow/pkg/entrypoint.Options.ExecuteProcess","level":"error","msg":"Process did not finish before 4h0m0s timeout","severity":"error","time":"2024-11-14T17:43:29Z"}
INFO[2024-11-14T17:43:29Z] Received signal.                              signal=interrupt 
```
Timed out on cluster deprovision => unlikely to be caused by this PR

https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oc/1914/pull-ci-openshift-oc-master-e2e-agnostic-ovn-cmd/1857056629762035712
```
: [sig-cli][Feature:LegacyCommandTests][Disruptive][Serial] test-cmd: test/cmd/images.sh [apigroup:image.openshift.io]
...
...error waiting for the pod 'test-cmd' to complete:  imagestreamtag tag-c --from-image=quay.io/openshifttest/hello-openshift:openshift' expecting failure...
```
This seems a bit more suspicious but there were recent failures like this ([last tuesday](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_oc/1911/pull-ci-openshift-oc-master-e2e-agnostic-ovn-cmd/1856295430619402240) ) so I'm not concerned.

/test e2e-agnostic-ovn-cmd e2e-aws-ovn-serial

**Comment by petr-muller** [comment]
/label tide/merge-method-squash

I have separated commits for easy review and to allow easy removing some changes if not desired for any reason, but I don't think we need them in commit history.

**Comment by ardaguclu** [comment]
/lgtm
/hold
In case you want additional review from your team

File: pkg/cli/admin/release/new.go
Comment (wking) [inline pkg/cli/admin/release/new.go:26]: Packages where the final directory name is different from their package name are annoying.  The explicit `digest` here is a way to underline that this package will be referred to as `digest` in the rest of the file, so folks don't have to wonder what `go-digest` picked internally as their name.  But :shrug:, I'll survive without the explicit local hint; it's not like folks wondering where `digest` is defined will have many search-hits in this file to sift through, so they'll probably be able to figure it out.

**Review by wking — APPROVED** [review]
/lgtm
/hold cancel

**Pull request merged**
